docs/parallel-workers.md

```diff
@@ -0,0 +1,8 @@
+---
+title: Parallel Workers
+custom_edit_url: https://github.com/stryker-mutator/stryker-js/edit/master/docs/parallel-workers.md
+---
+
+Stryker will always run checkers and test runners in parallel by creating worker processes (note, not `worker_threads`). The number of such processes forked is determined by the configuration option [`--concurrency`](./configuration.md#concurrency-number). 
+
+However, imagine running these parallel processes on a test suite which uses resources like a database connection, web server or file system. This means these processes can conflict if they write to the same database, file or utilize the same port. To solve this problem, Stryker sets a unique number (starting from 0 and incremented by 1) to each worker's environment variable `STRYKER_MUTATOR_WORKER`. This variable can be utilized to distribute the resources without conflicts - for instance, by setting the server port to `4444 + (+process.env.STRYKER_MUTATOR_WORKER)` instead of `4444`. 
```

Comment:
I think it would be awesome to document a small but complete example here using `jest`. Including:

- A tiny app that creates 2 simple endpoints. (i.e. one that echoes whatever is sent and a health check that responds with 'OK'). 
- A test file with `jest` that starts a server in `beforeAll`, closes it in `afterAll`, and has 3 `it`s. 2 its for the echo and 1 for the health check endpoint, without using the environment variable first. Maybe make sure that 1 mutant should survive somehow.

Run this example with concurrent runners in StrykerJS and show precisely the problem with a screenshot of the HTML report. Namely, the mutation score is 100%, and the error message complains about a port being unavailable.

Then show the changes to make with a git diff sample:

```diff
let app;
beforeAll(() => {
-  app = new MyApp(4444);
+  app = new MyApp(4444 + (+process.env.STRYKER_MUTATOR_WORKER || 0);
});
```

And then show what the report looks like after.

Do you think you're up for this challenge? If your example works, we can also make it an actual `e2e` test that runs inside our CI pipeline.

Reply:
Hm, you're right - this would be a nice addition. I think I can attempt to add this.